go/stats/rates.go

```diff
@@ -76,13 +79,16 @@ func NewRates(name string, countTracker CountTracker, samples int, interval time
 	if interval < 1*time.Second && interval != -1*time.Second {
 		panic("interval too small")
 	}
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
I think design wise it's better to pass in the context explicitly as a more idiomatic Go design. But that might mean many more changes, so dunno if that's desired?

Reply:
I agree, but it will require other areas to be refactored too: some of the functions that call `NewRates` themselves don't have contexts passed in and we might need to store some of the contexts around so that the cancellation code paths have access to them.

Hence thought I would take this quicker approach now.

Reply:
That's kinda what I suspected as well yeah.